src/scoring.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
+
+class Scoring {
+  constructor() {
+    this._frameScores = [];
+    this._cumulativeScores = [];
+  }
+
+  calculateScore(frames) {
+    for (const frame of frames) {
+      this._getFrameValues(frame, frames);
+      if (frame.rollOne() === null) {
+        break;
+      }
+      frame.rollTwo() === null
+        ? this._incompleteFrame(frame)
+        : this._completeFrame(frame);
+    }
+    return this._frameScores;
```

Comment:
This looks like a bit of a code smell. You return an instance variable from this method which is about calculating, which means you are probably using this instance variable to collate the score. However, instance variables should reflect a state of the object, not an intermediat result of a calculation.
Instead you can use othe methods and then pass the result inside their return... Something like:

Compare:
```js
let sum = 0

function add3(a, b, c) {
  add(a)
  add(b)
  add(c)
  return sum
}

function add(x) {
  sum += x
}
```
and
```js

function add3(a, b, c) {
  let sum = 0
  sum = add(a, sum)
  sum = add(b, sum)
  sum = add(c, sum)
  return sum
}

function add(x, y) {
  return x + y
}
```
This is a rather contrived example, but you can see how the sum variable is protected in the second way, within the add3 method, without risk of being misused, even though that function is still calling extra outside functions for helps in calculation.

Reply:
I think I understand this in general terms, but am a bit confused as to how to fix it in my code! The two main problems as I see them are the use of `this._frameScores` which is an array that at various points in the code gets scores pushed into it, and the use of `this._totalFrameScore`, which is more like the examples you cite (and is easier for me to wrap my head around - i.e. it should always equal the total frame score, not the intermediate stages of calculating that score).

Reply:
I think both should be handled locally. If other methods need to push things into it, then you can alway pass the array to them, and then return the new array from them.

eg:
```js

function aggregateThingsIntoArray() {
   let array = []
   array = addAThing(array, 34)
   return array
}

function addAThing(array, total, thing) {
    array.push(thing)
    return array
}
```
if you need to change several things at once, you can use destructuring assignment:
```js
function  {
   let array = []
   let total = 0
   [array, total] = addAThing(array, total, 34)
   return [array, total]
}

function addAThing(array, total, thing) {
    array.push(thing)
    total += thing
    return [array, total]
}
```
However, it's also possible there is an information duplication problem here too. Maybe you only need to figure out the array, and then the total score can be calculated based on the array.
```js
function  {
   let array = []
   array = addAThing(array, 34)
   total = array.reduce( (a, b) => a + b )
   return [array, total]
}
```

Reply:
thank you! I will see what I can do :)